tests/unit/models/transactions/test_set_oracle.py

```diff
@@ -0,0 +1,213 @@
+import time
+from unittest import TestCase
+
+from xrpl.models.exceptions import XRPLModelException
+from xrpl.models.transactions import OracleSet
+from xrpl.models.transactions.oracle_set import (
+    MAX_ORACLE_PROVIDER,
+    MAX_ORACLE_SYMBOL_CLASS,
+    MAX_ORACLE_URI,
+    PriceData,
+)
+
+_ACCOUNT = "rsA2LpzuawewSBQXkiju3YQTMzW13pAAdW"
+_PROVIDER = "chainlink"
+_ASSET_CLASS = "currency"
+
+_EMPTY_PROVIDER = ""
+_LENGTHY_PROVIDER = "X" * (MAX_ORACLE_PROVIDER + 1)
+
+
+class TestSetOracle(TestCase):
+    def test_valid(self):
+        tx = OracleSet(
+            account=_ACCOUNT,
+            oracle_document_id=1,
+            provider=_PROVIDER,
+            asset_class=_ASSET_CLASS,
+            last_update_time=int(time.time()),
+            price_data_series=[
+                PriceData(
+                    base_asset="XRP", quote_asset="USD", asset_price=740, scale=1
+                ),
+                PriceData(
+                    base_asset="BTC", quote_asset="EUR", asset_price=100, scale=2
+                ),
+            ],
+        )
+        self.assertTrue(tx.is_valid())
+
+    def validate_data_series(self):
+        # price_data_series is missing in the input
+        with self.assertRaises(XRPLModelException):
+            OracleSet(
+                account=_ACCOUNT,
+                oracle_document_id=1,
+                provider=_PROVIDER,
+                asset_class=_ASSET_CLASS,
+                last_update_time=int(time.time()),
+            )
+
+        # price_data_series exceeds the mandated length (10 elements)
+        with self.assertRaises(XRPLModelException):
```

Comment:
Each test should be a separate function.

Reply:
I feel it's more readable to have all the `price_data_series` tests under one function, instead of cluttering with so many tests

Reply:
fixed in [3957867](https://github.com/XRPLF/xrpl-py/pull/701/commits/3957867b12e9b76120b97bffd6a9d8d1b64f3cfa)